homeassistant/components/elmax/common.py

```diff
@@ -0,0 +1,229 @@
+        self._coordinator = coordinator
+        self._transitory_state = None
+
+    @property
+    def transitory_state(self) -> Any | None:
+        """Return the transitory state for this entity."""
+        return self._transitory_state
+
+    @transitory_state.setter
+    def transitory_state(self, value: Any) -> None:
+        """Set the transitory state value."""
+        self._transitory_state = value
+
+    @property
+    def panel_id(self) -> str:
+        """Retrieve the panel id."""
+        return self._panel.hash
+
+    @property
+    def unique_id(self) -> str | None:
+        """Provide a unique id for this entity."""
+        return self._device.endpoint_id
+
+    @property
+    def name(self) -> str | None:
+        """Return the entity name."""
+        return self._device.name
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return extra attributes."""
+        return {
+            "index": self._device.index,
+            "visible": self._device.visible,
+        }
+
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "identifiers": {(DOMAIN, self._panel.hash)},
+            "name": self._panel.get_name_by_user(
+                self._coordinator.http_client.get_authenticated_username()
+            ),
+            "manufacturer": "Elmax",
+            "model": self._panel_version,
+            "sw_version": self._panel_version,
+        }
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._panel.online
+
+    def _http_data_changed(self) -> None:
+        # Whenever new HTTP data is received from the coordinator we extract the stat of this
+        # device and store it locally for later use
+        device_state = self._coordinator.get_endpoint_state(self._device.endpoint_id)
+        if self._device is None or device_state.__dict__ != self._device.__dict__:
+            # If HTTP data has changed, we need to schedule a forced refresh
+            self._device = device_state
+            self.async_schedule_update_ha_state(force_refresh=True)
+
+        # Reset the transitory state as we did receive a fresh state
+        self._transitory_state = None
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass.
+
+        To be extended by integrations.
+        """
+        self._coordinator.async_add_listener(self._http_data_changed)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from hass.
+
+        To be extended by integrations.
+        """
+        self._coordinator.async_remove_listener(self._http_data_changed)
```

Comment:
I don't think this is needed, as this is handled by the DataUpdateCoordinator & CoordinatorEntity

Reply:
Yup. In #61273 I've extended a CoordinatorEntity rather than a simple Entity and removed such lines.